services/createFabricObjects.js

```diff
@@ -0,0 +1,365 @@
+      const { x, y, width, height } = offSet;
+
+      return new Promise(resolve => {
+        fabric.Image.fromURL(el.property.imageURL, img => {
+          img.set({
+            left: x + addNumber.dx,
+            top: y + addNumber.dy,
+          });
+
+          const mask = new fabric.Ellipse({
+            left: x + addNumber.dx,
+            top: y + addNumber.dy,
+            rx: width / 2,
+            ry: height / 2,
+            angle: el.property.arcData.endingAngle,
+            absolutePositioned: true,
+          });
+
+          img.scaleToWidth(width);
+          img.clipPath = mask;
+
+          resolve(img);
+        });
+      });
+    };
+
+    const result = await imageObject(number, position);
+
+    return result;
+  }
+
+  if (el.type === "RECTANGLE") {
+    const imageObject = async (addNumber, offSet) => {
+      const { x, y, width, height } = offSet;
+
+      return new Promise((resolve, reject) => {
+        const mask = new fabric.Rect({
+          left: x + addNumber.dx,
+          top: y + addNumber.dy,
+          width,
+          height,
+          rx: el.property?.cornerRadius,
+          ry: el.property?.cornerRadius,
+          absolutePositioned: true,
+        });
+
+        fabric.Image.fromURL(el.property.imageURL, img => {
+          try {
+            img.set({
+              left: x + addNumber.dx,
+              top: y + addNumber.dy,
+            });
+
+            img.scaleToWidth(width);
+            img.clipPath = mask;
+
+            resolve(img);
+          } catch {
+            reject(new Error("이미지 로드에 실패했습니다."));
+          }
+        });
+      });
+    };
+
+    const result = await imageObject(number, position);
+
+    return result;
+  }
+};
+
+const renderRect = async (el, number) => {
+  const isExistImg = el.property.fills[0]?.imageRef ?? null;
+
+  if (isExistImg) {
+    const imageFabricObject = await renderImg(el, number);
+
+    return imageFabricObject;
+  }
+
+  const style = el.property;
+  const bgColor = style.fills[0]?.color ?? style.backgroundColor ?? null;
+  const { strokes } = style;
+
+  return new fabric.Rect({
+    left: style.absoluteBoundingBox.x + number.dx,
+    top: style.absoluteBoundingBox.y + number.dy,
+    width: style.absoluteBoundingBox.width,
+    height: style.absoluteBoundingBox.height,
+    fill:
+      bgColor &&
+      `rgba(${bgColor.r * 255}, ${bgColor.g * 255}, ${bgColor.b * 255}, ${bgColor.a})`,
+    opacity: style.fills[0]?.opacity,
+    stroke: strokes.length
+      ? `rgba(${strokes[0].color.r * 255}, ${strokes[0].color.g * 255}, ${strokes[0].color.b * 255}, ${strokes[0].color.a})`
+      : 0,
+    strokeAlign: style.strokeAlign && style.strokeAlign,
+    rx: style.cornerRadius || 0,
+    ry: style.cornerRadius || 0,
+    visible: true,
+    evented: true,
+  });
+};
+
+const renderFrame = async (el, number) => {
+  const style = el.property;
+  const bgColor = style.fills[0]?.color ?? style.backgroundColor ?? null;
```

Comment:
- `P1`
```suggestion
  const backgoundColor = style.fills[0]?.color ?? style.backgroundColor ?? null;
```
축약어 사용은 최대한 지양해주시면 더욱 좋을 것 같습니다!